### PR TITLE
refactor: rewrited snackbar. allow stack

### PR DESCRIPTION
**Pull request by Ax52**
Refactor snackbar. Now it allows staking toasts. By default - up to 3 in a column.
Now, to use snackbar, just call `import { addToast } from "@/store"` and use it where you want (even outside reactive components / hooks!).

It allows 2 ways to use it:
1) Simple:
```js
addToast("Something were broken =-(")
```
// <=== NOTE: ===>

By default, severity is set to `"error"` to make it easier to handle rejections. But you can pass `"success"` | `"info"` | `"warning"` also as a second argument. Let's try:

```js
addToast("Hello World!", "success")
```

2) Custom:
```jsx
addToast({
  title: "Hello",
  message: "it's me!",
  severity: "info",
  button: <button>Close</button>,
})
```
for more info see JSDoc for the `addToast` function.

// <=== NOTE: ===>

Even in custom mode you can pass the severity as the second argument, and it will overwrite the severity, given in the options

### CODE REVIEW

**Review by glazkovpavel — APPROVED** [review]

File: packages/client/src/store/snackbarStore.ts [hunk 1]
```diff
@@ -1,79 +1,95 @@
...
+   * This is being set automatically unless specified manually.
+   */
+  key?: number;
+  /**
+   * Alert title
+   */
+  title?: string;
+  /**
+   * Alert message
+   */
```
Comment (itwassoclear) [inline packages/client/src/store/snackbarStore.ts:15]: А зачем комментарии в 3 строки?)

File: packages/client/src/utils/Snackbar.tsx [hunk 2]
```diff
@@ -1,33 +1,128 @@
...
+
+  const close = useCallback(() => {
+    setIsOpen(false);
+    setTimeout(() => {
+      closeToast(toast.key);
+    }, TIMEOUT);
+  }, [toast.key]);
+
+  const handleClose = (_: React.SyntheticEvent | Event, reason?: string) => {
+    if (reason === "clickaway") {
```
Comment (itwassoclear) [inline packages/client/src/utils/Snackbar.tsx:64]: if (reason === "clickaway") return;

**Review by itwassoclear — APPROVED** [review]
Так просто к тебе и не подокапываешься...

**Review by jodan13 — APPROVED** [review]

File: packages/client/src/store/snackbarStore.ts (hunk 1 above)
Comment (Ax52) [reply packages/client/src/store/snackbarStore.ts:15]: Попытки реализовать JSDoc аннотации ) Поменял

File: packages/client/src/utils/Snackbar.tsx (hunk 2 above)
Comment (Ax52) [reply packages/client/src/utils/Snackbar.tsx:64]: ок!

**Review by jodan13 — APPROVED** [review]

**Review by glazkovpavel — APPROVED** [review]

**Pull request merged**